x-pack/test/functional/services/ml/test_resources.ts

```diff
@@ -533,26 +534,57 @@ export function MachineLearningTestResourcesProvider({ getService }: FtrProvider
       log.debug('> ML saved objects deleted.');
     },
 
-    async installFleetPackage(packageIdentifier: string) {
-      log.debug(`Installing Fleet package '${packageIdentifier}'`);
+    async installFleetPackage(packageName: string) {
+      log.debug(`Installing Fleet package '${packageName}'`);
+
+      const version = await this.getFleetPackageVersion(packageName);
 
       await supertest
-        .post(`/api/fleet/epm/packages/${packageIdentifier}`)
+        .post(`/api/fleet/epm/packages/${packageName}-${version}`)
         .set(COMMON_REQUEST_HEADERS)
         .expect(200);
 
       log.debug(` > Installed`);
     },
 
-    async removeFleetPackage(packageIdentifier: string) {
-      log.debug(`Removing Fleet package '${packageIdentifier}'`);
+    async removeFleetPackage(packageName: string) {
+      log.debug(`Removing Fleet package '${packageName}'`);
+
+      const version = await this.getFleetPackageVersion(packageName);
 
       await supertest
-        .delete(`/api/fleet/epm/packages/${packageIdentifier}`)
+        .delete(`/api/fleet/epm/packages/${packageName}-${version}`)
         .set(COMMON_REQUEST_HEADERS)
         .expect(200);
 
       log.debug(` > Removed`);
     },
+
+    async getFleetPackageVersion(packageName: string): Promise<string> {
+      log.debug(`Fetching version for Fleet package '${packageName}'`);
+
+      const { body } = await supertest
+        .get(`/api/fleet/epm/packages?experimental=true`)
+        .set(COMMON_REQUEST_HEADERS)
+        .expect(200);
+
+      let packageVersion: string = '';
+
+      for (const packageDefinition of body.response) {
+        if (packageDefinition.name === packageName) {
+          if (packageDefinition.version) {
+            log.debug(` > found version '${packageDefinition.version}'`);
+            packageVersion = packageDefinition.version as string;
+            break;
+          }
+        }
+      }
```

Comment:
Nit, this could be written with less code.

```
for (const { name, version } of body.response) {
   if (name === packageName && version) {
     log.debug(` > found version '${version}'`);
     packageVersion = version as string;
     break;
  }
}
```
or even
```
const packageVersion = body.response.find(({name, version}) => name === packageName && version)?.version ?? ''
```

Reply:
Updated in aac44cc